sdk/cosmos/azure-cosmos/test/test_latest_session_token.py

```diff
@@ -0,0 +1,214 @@
+                                                                                                        True)
+
+        session_token = container.get_latest_session_token(feed_ranges_and_session_tokens, target_feed_range)
+        assert session_token == target_session_token
+        self.database.delete_container(container.id)
+
+
+    def test_latest_session_token_logical_hpk(self):
+        container = self.database.create_container("test_updated_session_token_from_logical_hpk" + str(uuid.uuid4()),
+                                                   PartitionKey(path=["/state", "/city", "/zipcode"], kind="MultiHash"),
+                                                   offer_throughput=400)
+        feed_ranges_and_session_tokens = []
+        previous_session_token = ""
+        target_pk = ['CA', 'LA1', '90001']
+        target_feed_range = container.feed_range_from_partition_key(target_pk)
+        target_session_token, previous_session_token = self.create_items_logical_pk(container, target_feed_range,
+                                                                                    previous_session_token,
+                                                                                    feed_ranges_and_session_tokens,
+                                                                                    True)
+        session_token = container.get_latest_session_token(feed_ranges_and_session_tokens, target_feed_range)
+
+        assert session_token == target_session_token
+        self.database.delete_container(container.id)
+
+
+    @staticmethod
+    def trigger_split(container, throughput):
+        print("Triggering a split in session token helpers")
+        container.replace_throughput(throughput)
+        print("changed offer to 11k")
+        print("--------------------------------")
+        print("Waiting for split to complete")
+        start_time = time.time()
+
+        while True:
+            offer = container.get_throughput()
+            if offer.properties['content'].get('isOfferReplacePending', False):
+                if time.time() - start_time > 60 * 25:  # timeout test at 25 minutes
+                    unittest.skip("Partition split didn't complete in time.")
+                else:
+                    print("Waiting for split to complete")
+                    time.sleep(60)
+            else:
+                break
+        print("Split in session token helpers has completed")
+
+    @staticmethod
+    def create_items_logical_pk(container, target_pk_range, previous_session_token, feed_ranges_and_session_tokens, hpk=False):
+        target_session_token = ""
+        for i in range(100):
+            item = create_item(hpk)
+            container.create_item(item, session_token=previous_session_token)
+            session_token = container.client_connection.last_response_headers[HttpHeaders.SessionToken]
```

Comment:
We can use the session token from the `response` as that is the best practice / thread-safe.

Reply:
changed